specs/APIReview_CustomDownload.md

```diff
@@ -0,0 +1,283 @@
+                       // Save path will be updated if a new one was provided.
+                       CHECK_FAILURE(args->put_SavePath(dialog.input.c_str()));
+                   }
+                   else if (dialog.canceled) {
+                       // If user selects `Cancel`, the download will be cancelled.
+                       CHECK_FAILURE(args->put_Cancel(TRUE));
+                   }
+                 return S_OK;
+             })
+             .Get(),
+         &m_DownloadStartingToken));
+    //! [DownloadStarting]
+}
+```
+# Remarks
+<!-- TEMPLATE
+    Explanation and guidance that doesn't fit into the Examples section.
+
+    APIs should only throw exceptions in exceptional conditions; basically,
+    only when there's a bug in the caller, such as argument exception.  But if for some
+    reason it's necessary for a caller to catch an exception from an API, call that
+    out with an explanation either here or in the Examples
+-->
+
+
+# API Notes
+<!-- TEMPLATE
+    Option 1: Give a one or two line description of each API (type and member),
+        or at least the ones that aren't obvious from their name. These
+        descriptions are what show up in IntelliSense. For properties, specify
+        the default value of the property if it isn't the type's default (for
+        example an int-typed property that doesn't default to zero.)
+
+    Option 2: Put these descriptions in the below API Details section,
+        with a "///" comment above the member or type.
+-->
+
+
+# API Details
+<!-- TEMPLATE
+    The exact API, in IDL format for our COM API and
+    in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/)
+    when possible, or in C# if starting with an API sketch for our .NET and WinRT API.
+
+    Include every new or modified type but use // ... to remove any methods,
+    properties, or events that are unchanged.
+
+    (GitHub's markdown syntax formatter does not (yet) know about MIDL3, so
+    use ```c# instead even when writing MIDL3.)
+
+    Example:
+
+    ```
+    /// Event args for the NewWindowRequested event. The event is fired when content
+    /// inside webview requested to open a new window (through window.open() and so on.)
+    [uuid(34acb11c-fc37-4418-9132-f9c21d1eafb9), object, pointer_default(unique)]
+    interface ICoreWebView2NewWindowRequestedEventArgs : IUnknown
+    {
+        // ...
+
+        /// Window features specified by the window.open call.
+        /// These features can be considered for positioning and sizing of
+        /// new webview windows.
+        [propget] HRESULT WindowFeatures([out, retval] ICoreWebView2WindowFeatures** windowFeatures);
+    }
+    ```
+
+    ```c# (but really MIDL3)
+    public class CoreWebView2NewWindowRequestedEventArgs
+    {
+        // ...
+
+	       public CoreWebView2WindowFeatures WindowFeatures { get; }
+    }
+    ```
+-->
+```
+/// Add an event handler for the DownloadStarting event.
```

Comment:
When the event handler fires does the download block/pause? Or does the download continue while the event handler runs?

Reply:
The download item is an interrupted "target pending" state - the actual data download occurs but is not saved until this is resolved. (Second bullet below)

// A regular DownloadItem (created for a download in this session of
// the browser) normally goes through the following states:
//      * Created (when download starts)
//      * Destination filename determined
//      * Entered into the history database.
//      * Made visible in the download shelf.
//      * All the data is saved.  Note that the actual data download occurs
//        in parallel with the above steps, but until those steps are
//        complete, the state of the data save will be ignored.
//      * Download file is renamed to its final name, and possibly
//        auto-opened.

Reply:
We need this sort of info in the docs. About when the download UI will show (the event blocks the download UI), and about what happens to the download.